tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -4939,23 +4939,28 @@ Value *CGMSHLSLRuntime::EmitHLSLLiteralCast(CodeGenFunction &CGF, Value *Src,
     }
   } else if (ConstantFP *CF = dyn_cast<ConstantFP>(Src)) {
     APFloat v = CF->getValueAPF();
-    double dv = v.convertToDouble();
     if (llvm::IntegerType *IT = dyn_cast<llvm::IntegerType>(DstTy)) {
+      APSInt iv;
+      bool isExact;
+      v.convertToInteger(iv, APFloat::roundingMode::rmTowardZero, &isExact);
```

Comment:
Not sure if `rmTowardZero` is correct. Is this defined for HLSL?

Reply:
Yea, that seems like it should be right for me.